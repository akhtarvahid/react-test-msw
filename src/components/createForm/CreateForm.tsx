import React from 'react'
import Form from 'react-bootstrap/Form';
import InputGroup from 'react-bootstrap/InputGroup';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';

type CreateFormType = {
}

const CreateForm: React.FC<CreateFormType> = () => {
  return (
    <Form>
      <Row className="justify-content-center">
        <Col sm={6}>
          <Col sm={12}>
            <Form.Group controlId="formName" className="mb-3">
              <Form.Label htmlFor="enterName">Enter name</Form.Label>
              <InputGroup className="mb-3">
                <Form.Control
                  aria-label="Enter name"
                  aria-describedby="inputGroup-sizing-default"
                  placeholder='Enter name'
                />
              </InputGroup>
            </Form.Group>
          </Col>
          <Col sm={12}>
            <Form.Group controlId="formCountry" className="mb-3">
              <Form.Label htmlFor="enterLocation">
                Select location
              </Form.Label>
              <Form.Select aria-label="Default select example">
                <option>Select country</option>
                <option value="india">India</option>
                <option value="uae">UAE</option>
                <option value="Turkey">Turkey</option>
              </Form.Select>
            </Form.Group>
          </Col>
          <Col>
            <Form.Label htmlFor="enterColor">
              Select favourite color
            </Form.Label>
            <Row sm={12}>
              <Col sm={3}><Form.Label htmlFor="exampleColorInput">Color picker</Form.Label></Col>
              <Col sm={9}>
                <Form.Control
                  type="color"
                  id="exampleColorInput"
                  defaultValue="#42bca8"
                  title="Choose your color"
                />
              </Col>
            </Row>
          </Col>
        </Col>
      </Row>
    </Form>
  )
}
export default CreateForm;