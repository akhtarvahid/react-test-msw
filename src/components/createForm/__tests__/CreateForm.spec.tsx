import { fireEvent, logRoles, render, screen } from '@testing-library/react';
import CreateForm from '../CreateForm';
import userEvent from '@testing-library/user-event';
import user from '@testing-library/user-event'
describe('CreateForm component tests', () => {
    const submitHandlerMock = jest.fn();

    test('sub components: Test1', () => {
        const view = render(<CreateForm colorSetter={submitHandlerMock} />);
        expect(view).toMatchSnapshot();
    });

    test('TextBox available: Test2', () => {
        render(<CreateForm colorSetter={submitHandlerMock} />);
        const nameInputElem = screen.getByRole('textbox', { name: 'Enter name' });
        expect(nameInputElem).toBeInTheDocument();

        //logRoles(nameInputElem)

        const nameInputLabel = screen.getByLabelText('Enter name');
        expect(nameInputLabel).toBeInTheDocument();
        //logRoles(nameInputLabel)

        const loactionSelectElem = screen.getByRole('combobox');
        expect(loactionSelectElem).toBeInTheDocument();
        //logRoles(loactionSelectElem)

        const colorInputElem = screen.getByLabelText('Color picker');
        expect(colorInputElem).toBeInTheDocument();
    });

    test('TextBox Change event: Test3', () => {
        render(<CreateForm colorSetter={submitHandlerMock} />);
        const nameInputElem = screen.getByRole('textbox', {
            name: "Enter name"
        });
        fireEvent.change(nameInputElem, { target: { value: 'xyz' } })
        expect(nameInputElem).toBeInTheDocument();
    });

    test('TextBox Location change event: Test4', () => {
        render(<CreateForm colorSetter={submitHandlerMock} />);
        const nameInputElem = screen.getByRole('combobox');
        fireEvent.change(nameInputElem, { target: { value: 'India' } })
        expect(nameInputElem).toBeInTheDocument();
    });
    test('TextBox Color change event: Test5', () => {
        render(<CreateForm colorSetter={submitHandlerMock} />);
        const colorInputElem = screen.getByRole('textbox', {
            name: "Color picker"
        });
        fireEvent.change(colorInputElem, { target: { value: 'abc' } })
        expect(colorInputElem).toBeInTheDocument();
    });

    test('TextBox Submit button: Test6', async () => {
        user.setup();
        render(<CreateForm colorSetter={submitHandlerMock} />);
        const submitBtn = screen.getByRole('button', {
            name: "Submit"
        });
        await user.click(submitBtn);
        expect(submitBtn).toBeInTheDocument();
        const colorInputElem = screen.getByRole('textbox', {
            name: "Color picker"
        });
        await userEvent.clear(colorInputElem);
        expect(colorInputElem).toHaveValue('');
        expect(submitHandlerMock).toHaveBeenCalledTimes(1);
    });
})
