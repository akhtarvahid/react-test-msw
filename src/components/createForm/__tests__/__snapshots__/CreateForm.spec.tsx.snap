// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateForm component tests sub components: Test1 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class=""
      >
        <div
          class="justify-content-center row"
        >
          <div
            class="col-sm-6"
          >
            <div
              class="col-sm-12"
            >
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="enterName"
                >
                  Enter name
                </label>
                <div
                  class="mb-3 input-group"
                >
                  <input
                    aria-describedby="inputGroup-sizing-default"
                    aria-label="Enter name"
                    class="form-control"
                    id="formName"
                    placeholder="Enter name"
                  />
                </div>
              </div>
            </div>
            <div
              class="col-sm-12"
            >
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="enterLocation"
                >
                  Select location
                </label>
                <select
                  aria-label="Default select example"
                  class="form-select"
                  id="formCountry"
                >
                  <option>
                    Select country
                  </option>
                  <option
                    value="india"
                  >
                    India
                  </option>
                  <option
                    value="uae"
                  >
                    UAE
                  </option>
                  <option
                    value="Turkey"
                  >
                    Turkey
                  </option>
                </select>
              </div>
            </div>
            <div
              class="col"
            >
              <label
                class="form-label"
                for="enterColor"
              >
                Select favourite color
              </label>
              <div
                class="row row-cols-sm-12"
              >
                <div
                  class="col-sm-3"
                >
                  <label
                    class="form-label"
                    for="exampleColorInput"
                  >
                    Color picker
                  </label>
                </div>
                <div
                  class="col-sm-9"
                >
                  <input
                    class="form-control form-control-color"
                    id="exampleColorInput"
                    title="Choose your color"
                    type="color"
                    value="#42bca8"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class=""
    >
      <div
        class="justify-content-center row"
      >
        <div
          class="col-sm-6"
        >
          <div
            class="col-sm-12"
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="enterName"
              >
                Enter name
              </label>
              <div
                class="mb-3 input-group"
              >
                <input
                  aria-describedby="inputGroup-sizing-default"
                  aria-label="Enter name"
                  class="form-control"
                  id="formName"
                  placeholder="Enter name"
                />
              </div>
            </div>
          </div>
          <div
            class="col-sm-12"
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="enterLocation"
              >
                Select location
              </label>
              <select
                aria-label="Default select example"
                class="form-select"
                id="formCountry"
              >
                <option>
                  Select country
                </option>
                <option
                  value="india"
                >
                  India
                </option>
                <option
                  value="uae"
                >
                  UAE
                </option>
                <option
                  value="Turkey"
                >
                  Turkey
                </option>
              </select>
            </div>
          </div>
          <div
            class="col"
          >
            <label
              class="form-label"
              for="enterColor"
            >
              Select favourite color
            </label>
            <div
              class="row row-cols-sm-12"
            >
              <div
                class="col-sm-3"
              >
                <label
                  class="form-label"
                  for="exampleColorInput"
                >
                  Color picker
                </label>
              </div>
              <div
                class="col-sm-9"
              >
                <input
                  class="form-control form-control-color"
                  id="exampleColorInput"
                  title="Choose your color"
                  type="color"
                  value="#42bca8"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
